src/app/ReadHandler.h

```diff
@@ -153,6 +153,11 @@ class ReadHandler : public Messaging::ExchangeDelegate
          * issues w.r.t the ReadHandler itself.
          */
         virtual ApplicationCallback * GetAppCallback() = 0;
+
+        /*
+         * Retrieve the InteractionalModelEngine that holds this ReadHandler.
+         */
+        virtual InteractionModelEngine * GetReadHandlerImEngine();
```

Comment:
```suggestion
        virtual InteractionModelEngine * GetInteractionModelEngine();
```

I assume we are not making this pure-virtual because there could conceivably be management callbacks that are not the IM engine right now and those would break?  I suspect in practice there aren't any and we could just make this pure virtual and have the override in IM engine...